src/WebJobs.Script/Rpc/WebHostLanguageWorkerChannelManager.cs

```diff
@@ -176,5 +187,36 @@ internal void AddOrUpdateWorkerChannels(RpcWebHostChannelReadyEvent rpcChannelRe
                         return existingLanguageWorkerChannels;
                     });
         }
+
+        public bool MatchingPlaceholderChannelExists(string selectedRuntime)
+        {
+            // Don't use placeholders if worker runtime not determined for function app
+            if (string.IsNullOrEmpty(selectedRuntime))
+            {
+                return false;
+            }
+
+            // Don't use placeholders if running dotnet
+            if (selectedRuntime.Equals(LanguageWorkerConstants.DotNetLanguageWorkerName, StringComparison.InvariantCultureIgnoreCase))
+            {
+                return false;
+            }
+
+            // Check environment config to see if placeholder worker can be used by the function app
+            if (_placeholderEnvironmentConfig.TryGetValue(selectedRuntime, out Dictionary<string, string> environmentConfig))
```

Comment:
- Move this to separate method
- Add unit tests
- Use this method in GetChannels i.e if condition does not match, return null from GetChannels to ensure, FunctionDispatcher does not get the channel that is about to shutdown